sycl/plugins/level_zero/pi_level_zero.hpp

```diff
@@ -236,6 +250,11 @@ struct _pi_context : _pi_object {
   // Keep the PI devices this PI context was created for.
   std::vector<pi_device> Devices;
 
+  // If devices in the context are sub-devices of the same device, we want to
+  // save a root-device.
+  pi_device RootDevice = nullptr;
```

Comment:
if we call this SingleRootDevice, and have it filled when context has just 1 device too, I think this will simplify semantics of this field, and also make other tests you are making simpler: you'd do something one for these context, and something else for multi-root-device contexts